meerk40t/core/elements.py

```diff
@@ -452,6 +454,40 @@ def __init__(self, kernel, index=None, *args, **kwargs):
         self._align_group = False
         self._align_stack = []
 
+        self._timing_stack = {}
+
+    def set_start_time(self, key):
+        if key in self._timing_stack:
+            self._timing_stack[key][0] = time()
+        else:
+            self._timing_stack[key] = [time(), 0, 0]
+
+    def set_end_time(self, key, display=True):
+        if key in self._timing_stack:
+            stime = self._timing_stack[key]
+            etime = time()
+            duration = etime - stime[0]
+            stime[0] = etime
+            stime[1] += duration
+            stime[2] += 1
+            if display:
+                # print (f"Duration for {key}: {duration:.2f} sec - calls: {stime[2]}, average={stime[1] / stime[2]:.2f} sec")
+                self.kernel._console_channel(
```

Comment:
This should probably have it's own channel rather than running things through console. If people want to open that channel and listen to the logging they could.